cmd/podman/main.go

```diff
@@ -36,6 +38,24 @@ func main() {
 		return
 	}
 
+	podmanshConst := "podmansh"
+	podmanshShell := "/bin/sh"
+	if filepath.Base(os.Args[0]) == podmanshConst || strings.HasSuffix(os.Args[0], filepath.Base(os.Getenv("SHELL"))) {
+		_, err := registry.NewContainerEngine(&cobra.Command{}, []string{podmanshConst, podmanshShell})
+		if err != nil {
+			logrus.Errorf("Error running NewContainerEngine: %w", err)
+			os.Exit(125)
+			return
+		}
+		err = containers.PodmanshWait(podmanshConst)
+		if err != nil {
+			logrus.Error(err)
+			os.Exit(125)
+			return
+		}
```

Comment:
There is a lot of special handling in the setup code that is missed like that. I am not very comfortable with just bypassing most of the logic. I would rather have this moved into the real exec() function. Besides setting os.Args maybe set another global podmansh variable that the exec() function can read and call PodmanshWait() then.

Reply:
How about we add a hidden option --wait 10 or something, then if podman exec sees the wait, it calls into the podmanshwait function.

Reply:
added a --wait option in a new commit. PTAL